Source/Scene/ShadowMap.js

```diff
@@ -1248,6 +1184,52 @@ define([
         }
     };
 
+    var scratchTexelStepSize = new Cartesian2();
+
+    ShadowMap.prototype.combineUniforms = function(uniforms, isTerrain) {
+        var bias = this._isPointLight ? this._pointBias : (isTerrain ? this._terrainBias : this._primitiveBias);
+
+        var that = this;
+        var mapUniforms = {
+            u_shadowMapTexture :function() {
+                return that._shadowMapTexture;
+            },
+            u_shadowMapTextureCube : function() {
+                return that._shadowMapTexture;
+            },
+            u_shadowMapMatrix : function() {
+                return that._shadowMapMatrix;
+            },
+            u_shadowMapCascadeSplits : function() {
+                return that._cascadeSplits;
+            },
+            u_shadowMapCascadeMatrices : function() {
+                return that._cascadeMatrices;
+            },
+            u_shadowMapLightDirectionEC : function() {
+                return that._lightDirectionEC;
+            },
+            u_shadowMapLightPositionEC : function() {
+                return that._lightPositionEC;
+            },
+            u_shadowMapCascadeDistances : function() {
+                return that._cascadeDistances;
+            },
+            u_shadowMapTexelSizeDepthBiasAndNormalShadingSmooth : function() {
+                var texelStepSize = scratchTexelStepSize;
+                texelStepSize.x = 1.0 / that._textureSize.x;
+                texelStepSize.y = 1.0 / that._textureSize.y;
+
+                return new Cartesian4(texelStepSize.x, texelStepSize.y, bias.depthBias, bias.normalShadingSmooth);
+            },
+            u_shadowMapNormalOffsetScaleDistanceAndMaxDistance : function() {
+                return new Cartesian3(bias.normalOffsetScale, that._distance, that._maximumDistance);
```

Comment:
We probably shouldn't be creating Cartesians all the time. Though this won't be a problem once all these uniforms are combined into structs for multiple shadow maps.